### ISSUE TITLE
Early Stopping in TimeSeriesPredictor?

### ISSUE THREAD

**Issue by DAD-FOX** [autogluon/autogluon#4757]
# Clarification on models Using Early Stopping by default in TimeSeriesPredictor

## Description  
I would like to understand which models in the `TimeSeriesPredictor` use early stopping by default. Additionally, could you please clarify how early stopping is implemented and configured in these models?  

### Key Questions  
1. **Default Usage**:  
   - Which models in `TimeSeriesPredictor` (e.g., Temporal Fusion Transformer, DeepAR, etc.) have early stopping activated by default?  

2. **Default Parameters**:  
   - Are there any default configurations for early stopping, such as:  
   - Patience (number of epochs to wait before stopping)?  
   - Minimum delta (the threshold for improvement)?  

4. **Customization**:  
   - How can I customize early stopping for models in `TimeSeriesPredictor`?  

Thank you for your assistance!

**Comment by shchur** [comment]
Hi, as of v1.2 all the deep learning models have early stopping enabled by default. This behavior is controlled using the `early_stopping_patience` hyperparameter (defaults to 20, see [docs](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-model-zoo.html#deep-learning-models)). The parameters of the `EarlyStopping` callback are set to their default values ([code](https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py#L516-L517)).

You can disable early stopping early stopping or change this behavior using the `early_stopping_patience` hyperparameter:
```python
predictor = TimeSeriesPredictor(...)
predictor.fit(..., hyperparameters={"DeepAR": {"early_stopping_patience": None}})  # disables early stopping
```

To use a custom early stopping strategy, you need to pass the custom callbacks to the using the `trainer_kwargs` hyperparameter like in the following example:
```python
from lightning.pytorch.callbacks import EarlyStopping

predictor.fit(..., hyperparameters={"DeepAR": {"trainer_kwargs": {"callbacks": [EarlyStopping(...)]}}})
```

Note that in any case the model with the best validation loss will be saved and used for prediction.
